tests/meltano/core/runner/test_runner.py

```diff
@@ -157,12 +157,12 @@ async def test_invoke(
 
     @pytest.mark.asyncio()
     @pytest.mark.parametrize(
-        ("full_refresh", "select_filter", "payload_flag"),
+        ("full_refresh", "merge_state", "select_filter", "payload_flag"),
         (
-            (False, [], Payload.STATE),
-            (True, [], Payload.STATE),
-            (False, ["entity"], Payload.STATE),
-            (True, ["entity"], Payload.INCOMPLETE_STATE),
+            (False, True, [], Payload.INCOMPLETE_STATE),
+            (True, True, [], Payload.INCOMPLETE_STATE),
+            (False, False, ["entity"], Payload.STATE),
+            (True, False, ["entity"], Payload.INCOMPLETE_STATE),
```

Comment:
Might be worth testing the new functionality with more test cases rather than overwriting the old ones

```suggestion
            (False, False, [], Payload.STATE),
            (True, False, [], Payload.STATE),
            (False, False, ["entity"],Payload.STATE),
            (True, False, ["entity"], Payload.INCOMPLETE_STATE),
            (False, True, [], Payload.INCOMPLETE_STATE),
            (True, True, [], Payload.INCOMPLETE_STATE),
            (True, True, ["entity"], Payload.INCOMPLETE_STATE),
```